### ISSUE TITLE
Cross signing keys aren't shared between sessions

### ISSUE THREAD

**Issue by alistair23** [matrix-org/dendrite#2360]
### Background information

- **Dendrite version or git SHA**: 073972646fd38b6987c2b62f8db5500c2f87d2d0
- **Monolith or Polylith?**: Monolith
- **SQLite3 or Postgres?**: Postgres
- **Running in Docker?**: No
- **`go version`**:
- **Client used (if applicable)**: Element Web and Desktop

### Description

 - **What** is the problem: Cross signing keys aren't shared between sessions
 - **Who** is affected: Server users
 - **How** is this bug manifesting:

I have two Element sessions. Each one says that it's verified and that the other session doesn't support encryption in the "Security and Privacy" section.

Keys are not syned between the two sessions, I can only decrypt my or others messages from one session

 - **When** did this first appear:

When updating from 0.6.5 to 0.8.1

I have tried resetting my cross signing keys, but that hasn't helped

### Steps to reproduce

- Clear all cross signing keys
- Try to chat in an encrypted room from two Element sessions

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from server logs, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.

Alternatively, please send logs to @kegan:matrix.org or @neilalexander:matrix.org
with a link to the respective Github issue, thanks!
-->

**Comment by alistair23** [comment]
Now I don't seem to be able to verify any sessions. I have just a single Element account logged in, and it sees itself as unverified

**Comment by alistair23** [comment]
It now seems like resetting keys isn't working. So I have reset them, but the people I'm talking to aren't being notified so they are still using old keys

**Comment by neilalexander** [comment]
A bunch of fixes for cross-signing went in with #2368, can you retry on the latest `main` commit?

**Comment by alistair23** [comment]
After resetting everything again and signing out of everything it seems to be working now on the latest main commit

**Comment by alistair23** [comment]
Ah nope, re-opening this.

So with latest master I have one Element sessions that works (before that nothing worked) but it's still unable to sync between my other sessions. It reports my other sessions as encryption not supported

**Comment by alistair23** [comment]
Back to unable to decrypted received messages

**Comment by alistair23** [comment]
Seems to be fixed in `0.8.2`
